---
title: Meta-analyses of MDD Symptom GWASs
author: Mark adams
output:
  html_document:
    df_print: kable
---

Meta analyze MDD Symptom GWASs for the following sets of cohorts:

1. Clinical/Enriched:
  - [PGC MDD](pgc-symptom-gwas.html) cases
  - [Australian of Depression Study](https://www.geneticsofdepression.org.au/)
  - [Generation Scotland](http://generationscotland.org) MDD cases
2. Community cohorts:
  - [ALSPAC](http://www.bristol.ac.uk/alspac/)
  - [Estonian Biobank](https://genomics.ut.ee/en/content/estonian-biobank)
  - [UK Biobank](https://www.ukbiobank.ac.uk)
3. All cohorts together

```{r symptoms}

library(readr)
library(dplyr)
library(readxl)
library(stringr)

dsm_mdd_symptoms_reference <-
read_delim("
MDD1;depressed;Depressed mood most of the day, nearly every day
MDD2;anhedonia;Markedly diminished interest or pleasure in all, or almost all, activities most of the day, nearly every day
MDD3;weightChange;Significant change in weight or appetite
MDD3a;weightLoss;Significant weight loss or decrease in appetite
MDD3b;weightGain;Significant weight gain or increase in appetite
MDD4;sleepChange;Sleeping too much or not sleeping enough
MDD4a;sleepProb;Insomnia nearly every day
MDD4b;sleepMore;Hypersomnia nearly every day
MDD5;psychomotor;Changes in speed/amount of moving or speaking
MDD5a;psychomotorFast;Psychomotor agitation nearly every day
MDD5b;psychomotorSlow;Psychomotor retardation nearly every day
MDD6;fatigue;Fatigue or loss of energy nearly every day
MDD7;worthless;Feelings of worthlessness or excessive or inappropriate guilt
MDD8;concentration;Diminished ability to think or concentrate, or indecisiveness
MDD9;death;Recurrent thoughts of death or suicide or a suicide attempt or a specific plan for attempting suicide
", col_names=c('Reference', 'Symptom Abbrev.', 'Description'), delim=';')

dsm_mdd_symptoms_reference

```


# Sumstats alignment

Align all sumstats to the reference panel used for the PGC MDD cohorts in preparation to run the [RICOPILI post-imputation command](https://docs.google.com/document/d/1o4bN_uLK4IEItXCSdeQkXfZwEpLuSCWlveJevRogi08/edit#heading=h.4008addvumol). 

Set up imputation reference directory

```{bash reference_info, eval=FALSE}

cd meta

impute_dirsub --refdir $(cat $HOME/README.hrcrel11 | grep '^/home/pgcdac/.*/HRC_reference\.r1-1') --out meta --reference_info

```

Alignment workflow

```{sh, eval=FALSE}
ln -s $(cat $HOME/README.hrcrel11 | grep '^/home/pgcdac/.*/HRC_reference\.r1-1') meta/HRC_reference.r1-1
nextflow run meta/align.nf -resume \
-work-dir /nfs/scratch/nfx-Go9q2ZbSjyfQrSXU/work \
-config lisa.config \
--out meta/aligned \
--frq="meta/HRC_reference.r1-1/HRC.r1-1.EGA.GRCh37.chr*.impute.plink.EUR.frq2.gz" \
--daner="sumstats/**/*daner*gz"
```

Number of SNPs pre- and post-alignment
```{r aligned}

aligned_count_files <- list.files('meta/aligned', pattern='snpcount', full.names=TRUE)

bind_rows(lapply(aligned_count_files, read_tsv, col_types=cols(gwas = col_character(), pre_align = col_integer(), post_align = col_integer())))

```

# Meta-analysis

Set up dataset lists for meta-analysis. First, create symlinks to all of the aligned sumstats files in the `meta` directory

```{bash symlink, eval=FALSE}

for gz in meta/aligned/*.gz; do
        ln -sr $gz meta/$(basename $gz);
done

```

Open mapping between sumstats filenames and MDD symptom reference numbers. Meta-analyze PGC+ADGS and ALSPAC+UKB.

```{r metafile}

# open file
cohorts <- read_tsv('meta/cohort_alignment.txt')

aligned_daner_files <- list.files('meta/aligned', pattern='gz', full.names=TRUE)

# create aligned filenames
cohorts_reference <-
cohorts %>%
left_join(dsm_mdd_symptoms_reference, by=c('reference'='Reference')) %>%
mutate(metagroup=case_when(cohort %in% c('AGDS', 'PGC', 'GenScot', 'janpy') ~ 'Clin',
                           cohort %in% c('ALSPAC', 'UKBB', 'EstBB') ~ 'Comm',
                           TRUE ~ NA_character_)) %>%
mutate(meta=paste0(metagroup, '.', reference, '_', `Symptom Abbrev.`, '.meta')) %>%
mutate(aligned_gz=str_replace(filename, 'gz', 'align.gz')) %>%
filter(aligned_gz %in% basename(aligned_daner_files)) %>%
# discard non-directional symptoms
filter(!reference %in% c('MDD3', 'MDD4')) %>%
# select meta-analyses where there are at least two sumstats
# contributing
group_by(meta) %>%
ungroup()

for(metafile in unique(cohorts_reference$meta)) {
	
	meta_cohorts <- cohorts_reference %>%
	  filter(meta == metafile) %>%
	  select(aligned_gz)

 	 write_tsv(meta_cohorts, file=file.path('meta', metafile), col_names=F)

}

```

Submit Ricopili post imputation command if it isn't already running and the final meta-analyzed daner file doesn't exist. 

```{bash postimp_navi, eval=FALSE}

cd meta

for metafile in *.meta; do
       outname=$(basename $metafile .meta)
       if [ ! -f j._pi_${outname} -a ! -f distribution/${outname}/daner_${outname}.gz ]; then
         echo postimp_navi --result $metafile --nolahunt --noldsc --popname EUR  --out $outname 
       fi
done

```

Concatenate all sets of cohorts together and run meta-analysis

```{bash postimp_navi_all, eval=FALSE}

cd meta

for mdd in MDD1_depressed MDD2_anhedonia MDD3a_weightLoss MDD3b_weightGain MDD4a_sleepProb MDD4b_sleepMore MDD5a_psychomotorFast MDD5b_psychomotorSlow MDD6_fatigue MDD7_worthless MDD8_concentration MDD9_death; do

	cat *.${mdd}.meta > ALL.${mdd}.meta4
	
done

for metafile in ALL.*.meta4; do
	   outname=$(basename $metafile .meta4)
	   if [ ! -f j._pi_${outname} ]; then
	   postimp_navi --result $metafile --nolahunt --noldsc --popname EUR  --out $outname
	   fi
done

```

```{r}

daner_xls <- list.files(here::here("meta/distribution"), pattern = "*.p4.clump.areator.sorted.1mhc.xls", full.names = TRUE, recursive = TRUE)
names(daner_xls) <- str_replace(basename(daner_xls), ".gz.p4.clump.areator.sorted.1mhc.xls", "")

daner_list <- lapply(daner_xls, function(xls) {
	read_excel(xls) |>
		mutate(ngt=as.numeric(ngt)) |>
		rename(FRQ_A=starts_with("FRQ_A"),
		       FRQ_U=starts_with("FRQ_U"))
	}
)

daner_hits <- bind_rows(daner_list, .id="filename") |>
filter(P <= 5e-8, !str_detect(filename, fixed("het")))

daner_hets <- bind_rows(daner_list, .id="filename") |>
filter(P <= 5e-8, str_detect(filename, fixed("het")))
```

Associations:

```{r}

alpha <- 5e-8 / (length(daner_xls) / 2)

daner_hits |>
	filter(P <= alpha |  gwas_catalog_span.6 != "-" | `genes.6.50kb(dist2index)` != "-") |>
	select(filename, SNP, P, gwas_catalog_span.6, `genes.6.50kb(dist2index)`) |>
	arrange(P)
```