--
title: PGC Symptom GWAS 
author: Bradley Jermy, Mark Adams 
output:
  html_document:
    toc: TRUE
    code_folding: hide
    number_sections: TRUE
    df_print: kable
    keep_md: true
  md_document:
    variant: markdown_github
---

Run GWAS of MDD symptoms on the [LISA server](http://geneticcluster.org/). Because these commands need to be run on a specific cluster with specific group permissions, this notebook documents the commands that were run rather than reproducing the whole analysis internally, with `eval=FALSE`.

# Directory setup

Softlink to all MDD files. Replace `PATH` with the actual path to the MDD Wave2 directory (listed in `README.mdd2sum`):

```{bash, eval=FALSE}
mkdir mdd_symptoms

cd mdd_symptoms

ln -s /home/PATH/* .
```

# Install [Ricopili](https://sites.google.com/a/broadinstitute.org/ricopili/)

```{bash, eval=FALSE}
#Install and unpack ricopili - follow this link for custom installation https://docs.google.com/document/d/14aa-oeT5hF541I8hHsDAL_42oyvlHRC5FWR7gir4xco/edit#heading=h.y8igfs7neh22
wget https://sites.google.com/a/broadinstitute.org/ricopili/download//rp_bin.2019_Jun_25.001.tar.gz

tar -xvzf rp_bin.2019_Jun_25.001.tar.gz

#Create custom installation file in ricopili

#TEST!!!  Postimp RICOPILI code - run this home directory after all softlinks have been set-up
postimp_navi --out MDD1_case_con --mds MDD29.0515.nproj.menv.mds_cov --coco 1,2,3,4,5,6 --refiex refiex.mddw2v01
```

# Symptom phenotype files

Launch an interactive session

```{bash,  eval=FALSE}

#Go into interactive mode
srun -n 16 -t 1:00:00 --pty bash -il

#Load R

module load 2022
module load R/4.2.1-foss-2022a

R

```

Load libraries and configure input directories

```{r}
library(yaml)
library(readxl)
library(readr)
library(stringr)
library(dplyr)
library(tidyr)

# load configuration file (see config-example.yaml')
config <- yaml.load_file('config.yaml')

# input and output directories and files
mdd_v1_dir <- config$data$pgc$mdd$v1
mdd_v2_dir <- config$data$pgc$mdd$v2

output_dir = "pgc_gwas"
```

## Phenotype files 

Extract 12 MDD symptoms

```{r}

#Extract 12 symptoms
symptoms <- c("MDD1", "MDD2", "MDD3a", "MDD3b", "MDD4a", "MDD4b", "MDD5a", "MDD5b", "MDD6", "MDD7", "MDD8", "MDD9")
studies <- c("RADIANT", "STR", "ROTTERDAM", "QIMR", "NESDANTR", "GSK", "GENRED", "GENRED2", "CoLaus", "COFAMS", "BonnMH")
studies_v2 <- c("BiDirect")
study_noupdate <- c("STARD", "SHIP0", "MARS", "GENPOD")

dir.create(output_dir, recursive=TRUE)

pheno_xls <- c(
  list.files(file.path(mdd_v1_dir, "secondary_phenotypes", "1216_update"), ".xls", full.names=TRUE),
  list.files(file.path(mdd_v2_dir, "phenotype"), ".xls", full.names=TRUE)
)

names(pheno_xls) <- basename(pheno_xls)

phenotypes <- 
bind_rows(
lapply(pheno_xls, 
  function(xls) read_excel(xls) |>
                       mutate(across(ID2:Sub.study, .fns=as.character)) |>
                       select(ID1, ID2, Study, Sub.study, MDD1:MDD9) |> 
                       mutate(across(MDD1:MDD9, as.numeric))
),
.id="filename"
) |>
separate(filename, into=c('file', 'update'))
```

## Code phenotypes

```{r}
pheno_au <- 
phenotypes |> mutate(across(MDD1:MDD9, ~ case_when(.x == 1 ~ 2L,
                                             .x == 0 ~ 1L,
                                             TRUE ~ NA_integer_))) |>
mutate(dataset = sapply(str_split(ID1, pattern = "_", n = 4), function(id) id[3]))

write_tsv(pheno_au, file.path(output_dir, "mddw2_symptoms.txt"))
write_tsv(select(pheno_au, FID=ID1, IID=ID2, starts_with("MDD")), file.path(output_dir, "mddw2_symptoms.pheno"), na="-9")
```

## Covariates

```{r}

# extract covariates
v1_cov <- read_table(file.path(mdd_v1_dir, "MDD29.0515.nproj.menv.mds_cov"))
v2_cov <- read_table(file.path(mdd_v2_dir, "cobg_gw.Bidirect.menv.mds_cov"))
covar <- bind_rows(v1_cov, v2_cov)

write_tsv(covar, file.path(output_dir, "mddw2.covar"))
```

## Cases cohorts

Identify cohorts with cases that vary in symptoms.

```{r}

cases <- pheno_au |>
filter(str_detect(ID1, "^cas_mdd"))

cohorts_with_case_symptoms <- 
cases |>
pivot_longer(cols=MDD1:MDD9, names_to = "symptom", values_to = "status") |>
filter(status %in% 1:2) |>
distinct(dataset, symptom, status) |>
count(dataset, symptom) |>
filter(n == 2) |>
distinct(dataset)

cases_ids <- cases |>
filter(dataset %in% pull(cohorts_with_case_symptoms, dataset)) |>
select(FID=ID1, IID=ID2)

write_tsv(cases_ids, file.path(output_dir, "cases.id"))
write_tsv(cohorts_with_case_symptoms, file.path(output_dir, "cohorts.keep"), col_names=F)
```

## UKB Overlap

```{r}
ukb <- read_table("pgc_gwas/959_PGC_UKB_overlap.id", col_names=c("FID", "IID"))
```

# GWAS Pipeline

Run GWAS across all symptoms for just cases

```{sh, eval=FALSE}

# symlink hardcall filenames

mkdir -p pgc_gwas/cobg_dir_genome_wide

V1=$(cat config.yaml | grep "v1: /" | awk '{print $2}')
V2=$(cat config.yaml | grep "v2: /" | awk '{print $2}')

while read cohort; do
  if [ -f $V1/cobg_dir_genome_wide/mdd_${cohort}_eur_sr-qc*.hg19.ch.fl.bgn.bed ]; then
    echo Linked $cohort
    ln -sf $V1/cobg_dir_genome_wide/mdd_${cohort}_*.bgn.{bed,bim,fam} pgc_gwas/cobg_dir_genome_wide/
 elif [ -f $V1/genred_case_only/mdd_${cohort}_eur_sr-qc*.hg19.ch.fl.bgn.cases.bed ]; then
 	echo Linked $cohort
 	ln -sf $V1/genred_case_only/mdd_${cohort}_*.bgn.cases.{bed,bim,fam} pgc_gwas/cobg_dir_genome_wide/
  elif [ -f $V2/cobg_dir_genome_wide/mdd_${cohort}_eur_sr-qc2.hg19.ch.fl.bgn.bed ]; then
    echo Linked $cohort
    ln -sf $V2/cobg_dir_genome_wide/mdd_${cohort}_*.bgn.{bed,bim,fam} pgc_gwas/cobg_dir_genome_wide/
  else
    echo Did not link $cohort
  fi
done<pgc_gwas/cohorts.keep

# patch in cof3 (cof2) rau2 (raus)
for cohort in cof3 rau2; do
  if [ -f $V1/cobg_dir_genome_wide/mdd_${cohort}_eur_sr-qc*.hg19.ch.fl.bgn.bed ]; then
    echo Linked $cohort
    ln -sf $V1/cobg_dir_genome_wide/mdd_${cohort}_*.bgn.{bed,bim,fam} pgc_gwas/cobg_dir_genome_wide/
  else
    echo Did not link $cohort
  fi
done
```

```{sh, eval=FALSE}

nextflow run pgc-symptom-gwas.nf \
-config lisa.config \
-work /nfs/scratch/nfx-qqZbmwsao3qBufyNACW/work
-resume \
--bfile="pgc_gwas/cobg_dir_genome_wide/mdd_*.bgn*{bed,bim,fam}"

```